### PR TITLE
Error for annotating void methods annotated with @Query

### LINKED ISSUE #348: Validation for GraphQL `@Query` methods with `void` return type
See https://download.eclipse.org/microprofile/microprofile-graphql-1.0/microprofile-graphql.html#_void_queries

### PR DESCRIPTION
**Pull request by datho7561**
Closes #348

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/graphql/java/MicroProfileGraphQLASTValidator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * Diagnostics for microprofile-graphql.
+ *
+ * @see https://download.eclipse.org/microprofile/microprofile-graphql-1.0/microprofile-graphql.html
+ */
+public class MicroProfileGraphQLASTValidator extends JavaASTValidator {
+
+	private static final Logger LOGGER = Logger.getLogger(MicroProfileGraphQLASTValidator.class.getName());
+
+	private static final String NO_VOID_MESSAGE = "''{0}'' is annotated with microprofile-graphql''s @Query, so it cannot have ''void'' as a return type.";
```
Comment (JessicaJHee) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/graphql/java/MicroProfileGraphQLASTValidator.java:46]: Would it make more sense to have the message be something more simplistic and straightforward like: 
Methods annotated with microprofile-graphql''s @Query cannot have ''void'' as a return type

**Review by JessicaJHee — COMMENTED** [review]
Works really well! Just had one small suggestion about the error message.

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/graphql/java/MicroProfileGraphQLASTValidator.java (hunk 1 above)
Comment (datho7561) [reply microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/graphql/java/MicroProfileGraphQLASTValidator.java:46]: Yes, I like that wording much better

**Review by JessicaJHee — APPROVED** [review]
LGTM, thank you!

**Pull request merged**